internal/schemautil/ipfilter.go

```diff
@@ -7,29 +7,29 @@ import (
 // NormalizeIpFilter compares a list of IP filters set in TF and a sorted version coming
 // from Aiven and takes sort IP filters such that all matching entries will be in
 // the same order as defined in the TF manifest.
-func NormalizeIpFilter(tfUserConfig interface{}, userConfig []map[string]interface{}) []map[string]interface{} {
+func NormalizeIpFilter(tfUserConfig interface{}, userConfig []map[string]interface{}) {
 	tfInt, ok := tfUserConfig.([]interface{})
 
 	if len(userConfig) == 0 {
-		return userConfig
```

Comment:
what was the need to get rid of the idiomatic Go way to modify data?

although, I do agree that this would need some extra code which copies the map

Reply:
You don't create new map, you modify existing https://go.dev/play/p/TjqI6b4VDjL. Returning a value looks like you return a new value, but it's not.

Reply:
I understand how maps function; as I said above, this would require to add extra code which copies the map, e.g. by applying `maps.Clone`, so I do agree that the code does not works as expected the way we have it at the moment